pkg/operator/controller/ingress/status_test.go

```diff
@@ -407,10 +408,28 @@ func TestComputeIngressDegradedCondition(t *testing.T) {
 			expectIngressDegradedStatus: operatorv1.ConditionFalse,
 			expectRequeue:               false,
 		},
+		{
+			name: "default ingress controller, canary check failing",
+			conditions: []operatorv1.OperatorCondition{
+				cond(IngressControllerCanaryCheckSuccessConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Minute*-1)),
```

Comment:
This time value appears to be right on the earlier edge of the grace period.  Initially, I was surprised this test case passed, and I suspect it could be racy, but `TestComputeIngressDegradedCondition` doesn't use a fake clock (maybe it should?), and I guess enough time passes (even if it's only a few microseconds) that the the grace period has elapsed at the point in time when the test case is evaluated.  I'd feel more comfortable with this test case though if the last transition time were initialized to at least 1 second further in the past than the earlier edge of the grace period:
```suggestion
				cond(IngressControllerCanaryCheckSuccessConditionType, operatorv1.ConditionFalse, "", clock.Now().Add(time.Second * -61)),
```

Reply:
Fixed, Thanks for catching this!